### PR TITLE
Make cProfile the default where available (Python 3) and add support for PyPy

### PR DESCRIPTION
**Pull request by Flamefire**
The overhead of cProfile is much lower and there are no downsides, so make it the default for Python 3 (not available for Python 2)

### CODE REVIEW

File: scorep/__main__.py [hunk 1]
```diff
@@ -22,7 +22,10 @@ def scorep_main(argv=None):
     no_default_threads = False
     no_default_compiler = False
     no_instrumenter = False
-    instrumenter_type = "profile"
+    if sys.version_info.major >= 3:
```
Comment (AndreasGocht) [inline scorep/__main__.py:25]: The simple version check does not work for interpreters like pypy . By the way: the same is true for the c bindings in general. The profile interface works with pypy, but not the cProfile.

**Review by AndreasGocht — COMMENTED** [review]
Do we wan't to address the cProfile does not work for pypy thing?

Best,

Andreas

File: scorep/__main__.py (hunk 1 above)
Comment (Flamefire) [reply scorep/__main__.py:25]: I could make this more elaborate, e.g. try-import. However currently we have the same condition in setup.py, isn't that used for pypy too?

**Comment by Flamefire** [comment]
> Do we wan't to address the cProfile does not work for pypy thing?

As it never can work for pypy I disabled it entirely for that.

**Comment by AndreasGocht** [comment]
Thanks. Looks good.

Best,

Andreas

**Pull request merged**